### ISSUE TITLE
Coder fails to retrieve workspaces - Database error

### ISSUE THREAD

**Issue by fpoussin** [coder/coder#4508]
Opening due to #4359 still happening

Same problem but different error message:
```
{"message":"Internal error fetching workspaces.","detail":"get authorized workspaces: pq: more than one row returned by a subquery used as an expression"}
```

**Comment by kylecarbs** [comment]
Oofta, now that's odd. I'm happy it's different I guess, but likely related to the same cause. I'll investigate!

**Comment by kylecarbs** [comment]
@fpoussin we weren't filtering by the `deleted` property for users and templates, so deleted users with the same username would conflict on a query. This will be fixed once we do a release with the attached PR.

**Comment by kylecarbs** [comment]
@fpoussin this has been fixed and released in v0.9.10!
